Functional annotation and comparative genomics
====

This is the last module of the Microbial Genomics course at the MSc Bioinformatics and Biocomplexity at Utrecht University.

The **objectives** in this module are:
- to understand how the function of proteins are predicted
- to learn how to perform typical -omics data analyses to identify candidate genes for further functional analyses
- to think critically about the results obtained with bioinformatics tools
- to find relevant information

This final module of the course is more biological than previous ones and is therefore more prone to variability in the results. The answers to exercises are only for indication; there are other ways to perform the analyses which will yield different results (no better or worse, just different). In each exercise, you must use a critical mind and search for relevant information to decide what you think is sensible from a biological point of view.

** **

**Overview of the two-day module**

*Day 1: 26 Nov, BOL â€“ 1.138 / 1.075*
1. 9h-9h45 Introduction about fungal genomic research and functional annotation
2. Exercise 1: annotation using similarity search
3. Exercise 2: search for conserved domains
4. Exercise 3: search for conserved motifs
5. 15h15-16h Wrap-up session

*Day 2: 27 Nov, BOL 1.075 / 1.138*
1. 9h-9h30 Introduction about the biology of fungal plant pathogens
2. Exercise 1: orthologous protein families
3. Exercise 2: compare the gene content between different species
4. 11h15-11h45 Wrap-up session
5. Exercise 3: case study with expression data
6. Exercise 4: case study with secondary metabolite gene clusters
7. 15h15-16h Wrap-up session
8. Questions

** **

**Data availability**

On the server B-MBIMIGE2024 ~/data/fungalgenomics_collemare/day1/ and ~/data/fungalgenomics_collemare/day2/

All presentations, assignments, command lines and answers on https://github.com/JeromeCollemare/Microbial_Genomics_course_24day* in due time
